vm/vm.go

```diff
@@ -331,13 +330,16 @@ func (vm *VM) Initialize(
 
 	vm.mempool = mempool.New[*chain.Transaction](vm.tracer, vm.config.MempoolSize, vm.config.MempoolSponsorSize)
 
-	vm.builder = builder.NewTime(toEngine, snowCtx.Log, vm.mempool, func(t int64) (int64, int64, error) {
-		blk, err := vm.GetStatefulBlock(context.TODO(), vm.preferred)
-		if err != nil {
-			return 0, 0, err
-		}
-		return blk.Tmstmp, vm.ruleFactory.GetRules(t).GetMinBlockGap(), nil
-	})
+	// if we haven't had the builder created by the options, create it now.
+	if vm.builder == nil {
+		vm.builder = builder.NewTime(toEngine, snowCtx.Log, vm.mempool, func(t int64) (int64, int64, error) {
+			blk, err := vm.GetStatefulBlock(context.TODO(), vm.preferred)
+			if err != nil {
+				return 0, 0, err
+			}
+			return blk.Tmstmp, vm.ruleFactory.GetRules(t).GetMinBlockGap(), nil
+		})
+	}
```

Comment:
Why this check rather than moving builder initialization before applying options?

Alternatively, could we make this more obviously mutually exclusive by constructing these components inside of `applyOptions` regardless of if it is going to create `NewTime` or `NewManual` ?

Reply:
I think both are good ideas. I'll move the `applyOptions` past the initialization of the mempool, have the initialization of the builder done exclusively within the `applyOptions` and replicate the same of the `gossipier`